Add deCONZ component documentation

Kane610 opened the pull request:
**Description:**
Component supporting deCONZ REST API exposing Zigbee devices such as Lights and Sensors.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant):** home-assistant/home-assistant#10321

## Checklist:

  - [x] New documentation for platforms/components and features should go to `next`. 
  - [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

Kane610 committed:
Deconz hub documentation

Kane610 committed:
Add binary sensor documentation

Kane610 committed:
Add sensor documentation

Kane610 committed:
Add light documentation

Kane610 committed:
Improve description

Kane610 committed:
Add information about HASS events from ZHASwitch buttonpresses

Kane610 committed:
Added description about how entity id naming works

donnib commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,57 @@
+## {% linkable_title Configuration variables %}
+
+- **host** (*Required*): The IP address to your deCONZ server.
+- **api-key** (*Optional*): The username to your Axis device. Default is 'root'.
```
Left overs from the Axis component ;)

donnib commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,57 @@
+## {% linkable_title Device services %}
+Available services: `generate_api_key`.
+
+#### {% linkable_title Service `deconz/generate_api_key` %}
```
Please add a link to deconz documentation for this http://dresden-elektronik.github.io/deconz-rest-doc/getting_started/

donnib commented:
So here is some general feedback :

- I think it would be VERY beneficial to make a list of the real devices supported since users will ask questions like does THIS specific device work ? so make a list similar to the one i started :
https://bitbucket.org/mihaicph/deconz-custom-module/overview
- Add examples of an event automation like i did in above link
- Explain to users how they find the ID's in deconz by pointing to Postman for example (you can get some inspiration what i wrote)
- Make a list of things that you know doesn't work at the moment or won't work when released such as renames in deconz only after reboot or what id deconz is rebooted, does it then still work, such things.
- Also i think a best practice should be there such as explaining to users that they should try to make groups in deConz then control the groups when there is more than one light you wish to control at the same time, failing to do so can cause some weird Zigbee problems.

Kane610 committed:
Added links to dresden elektronik, deconz and deconz rest api
Added automation examples for a IKEA Trådfri dimmer and input number
Added list of supported device types
Added lists of verified to work devices
Removed references to Axis component

Kane610 committed:
Changed description of how to generate API key

Kane610 committed:
Added information about configuration variables username and password

simonporter007 commented on source/_components/light.deconz.markdown:
```diff
@@ -0,0 +1,23 @@
+
+#### {% linkable_title Verified to be supported sensors %}
+
+- IKEA Trådfri bulb GU10 W 400lm
```
Tested "TRADFRI bulb E27 WS�opal 980lm" as working. Not sure about the weird character in there but thats how it displays in the front end (and probably deconz to be fair)

Kane610 replied:
Hmm, maybe I need to decode it to a specific type like UTF-8?

simonporter007 commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,112 @@
+
+- **host** (*Required*): The IP address to your deCONZ server.
+- **api-key** (*Optional*): The API key to your deCONZ server.
+- **port** (*Optional*): Configure port deCONZ web server is accessible from. Default is 80.
```
I'm not sure if it's used here but in the more [recent versions](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/238) of deconz you can change the websocket port. Does this need to be taken into consideration here?

Kane610 replied:
Websocket port is taken directly from Deconz, no need to configure.

donnib commented on source/_components/binary_sensor.deconz.markdown:
```diff
@@ -0,0 +1,29 @@
+
+- Open/close detection
+- Presence detection
+  - IKEA Trådfri Motion Sensor
```
Add HUE Motion sensor

donnib commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,112 @@
+
+[deCONZ](https://www.dresden-elektronik.de/funktechnik/products/software/pc/deconz/) by [Dresden Elektronik](https://www.dresden-elektronik.de) is a software that communicates with Conbee/Raspbee Zigbee gateways and exposes Zigbee devices that are connected to the gateway.
+
+[deCONZ REST API](http://dresden-elektronik.github.io/deconz-rest-doc/) verified from v2.04.86.
```
Change to 2.04.89

donnib commented on source/_components/sensor.deconz.markdown:
```diff
@@ -0,0 +1,35 @@
+- Light level sensor
+- Pressure sensor
+- Switches
+  - IKEA Trådfri wireless dimmer
```
Add IKEA Trådfri remote http://www.ikea.com/us/en/catalog/products/20303317/
Add Xiaomi Humidity/Temperature sensor https://www.gearbest.com/access-control/pp_626702.html
Add Philips HUE Dim Switch http://www2.meethue.com/en-us/p/hue-dimmer-switch/046677458140
Add Xiaomi Cube https://www.gearbest.com/living-appliances/pp_364494.html?vip=3160986&gclid=CjwKCAiArrrQBRBbEiwAH_6sNOFBNv-r-QMC2huQ9ODXWxARRAJtGRdwWZO_1GEt9MAJg1IXLS1hlhoCDT4QAvD_BwE
Add Philips Motion Sensor http://www2.meethue.com/en-hk/p/hue-motion-sensor/8718696769881

Kane610 committed:
Updated lists of verified devices

Kane610 committed:
Links from list of supported device to sub modules

Kane610 committed:
Remove documentation for api_key_generation since functionality has been removed

Kane610 committed:
Add documentation for configure service

Kane610 committed:
Documentation for scenes

Kane610 committed:
Bumped HASS release version to 0.59

Kane610 committed:
Added Xiaomi Aqara Smart Wireless Switch to list of verified to work devices

Kane610 committed:
Added improved automation example by simonporter007

Kane610 committed:
Added information about deCONZ configurator

Kane610 committed:
Fix raw code

donnib commented on source/_components/light.deconz.markdown:
```diff
@@ -0,0 +1,24 @@
+
+- IKEA Trådfri Bulb GU10 W 400lm
+- OSRAM Flex RGBW
+- IKEA Trådfri Bulb E27 WS Opal 980lm
```
Please add following :

TRADFRI bulb E14 WS opal 400lm
TRADFRI bulb E27 opal 1000lm
OSRAM Gardenpole RGBW-Lightify

Kane610 committed:
Added more lights to supported list

Kane610 committed:
Make description about entity id same in all types

Kane610 committed:
Updated installation instructions

Kane610 committed:
Added Dresden Elektroniks logo

tseel commented:
Can confirm following devices to be working fine:

Xiaomi WXKG01LM Smart Home Wireless Switch
Xiaomi WXKG11LM Aqara Smart Wireless Switch
Xiaomi WXKG02LM Aqara Smart Light Switch
Xiaomi RTCGQ01LM Motion Sensor
Xiaomi RTCGQ01LM Smart Home Aqara Human Body Sensor
Xiaomi MCCGQ01LM Smart Home Security Door & Window Contact Sensor
Xiaomi WSDCGQ01LM MiJia Smart Temperature & Humidity Sensor
Xiaomi WSDCGQ11LM Aqara Temperature Humidity Sensor

Due to lack of support in deconz, battery values are still not reported.

Kane610 committed:
Added Tseels verified to work devices

Kane610 commented:
@tseel not any of those devices?

Kane610 committed:
Bump ha_release string

tseel commented:
@Kane610 It's a known issue in the rest api of deconz and announced for the future, see dresden-elektronik/deconz-rest-plugin#13

Kane610 committed:
Added information about groups
Added additional verified lights

Kane610 committed:
Link to information about switch not being an entity on main hub page

Kane610 committed:
Improved information about switches not being normal entities

Kane610 committed:
It should be api_key not api-key

Kane610 committed:
Removed specifying deconz version

frenck commented on source/_components/sensor.deconz.markdown:
```diff
@@ -0,0 +1,45 @@
+---
+layout: page
+title: deCONZ Sensor
+description: "Instructions how to integrate Zigbee sensors from deCONZ into Home Assistant."
```
`Instructions how to` -> `Instructions on how to`

frenck commented on source/_components/scene.deconz.markdown:
```diff
@@ -0,0 +1,18 @@
+---
+layout: page
+title: "deCONZ"
+description: "Instructions how to integrate deCONZ scenes into Home Assistant."
```
`Instructions how to` -> `Instructions on how to`

frenck commented on source/_components/scene.deconz.markdown:
```diff
@@ -0,0 +1,18 @@
+footer: true
+logo: deconz.jpeg
+ha_category: Scene
+ha_release: "0.60"
```
bump to `0.61`

frenck reviewed (approved):
@Kane610 Requested some small change via review. Could you take a look?

frenck reviewed (changes requested):
Click approve instead of request changes (correcting)

Kane610 committed:
Bump version to 0.61

Kane610 committed:
Fix typos

Kane610 committed:
Use configuration tags to describe configuration parameters

Kane610 commented:
@frenck thanks for feedback! I followed your suggested improvements.

frenck reviewed (dismissed):
Thanks @Kane610 🏅

Kane610 commented:
Much appreciated @frenck !

MartinHjelmare commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,138 @@
+- [Switches (Remote Controls)](/components/sensor/deconz/)
+- [Temperature Sensors](/components/sensor/deconz/)
+
+## {% linkable_title Configuration variables %}
```
This header is not needed. It will be defined automatically when using the `configuration` tags.

MartinHjelmare commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,150 @@
+
+| Parameter | Description                                             |
+|-----------|---------------------------------------------------------|
+| `field`   | String representing a specific device in deCONZ.        |
```
Please format as our standard service table. Also extend this table with info on if parameter is optional. See eg:
https://home-assistant.io/components/media_player.sonos/#service-sonos_snapshot

Kane610 committed:
Fix martinhjelamares comments

MartinHjelmare commented on source/_components/deconz.markdown:
```diff
@@ -0,0 +1,148 @@
+#### {% linkable_title Service `deconz/configure` %}
+Set attribute of device in Deconz using [Rest API](http://dresden-elektronik.github.io/deconz-rest-doc/rest/).
+
+| Parameter | Optional | Description |
```
Please rename `Parameter` to `Service data attribute`.

Kane610 committed:
Rename parameter service data attribute

MartinHjelmare reviewed (approved):
Thanks!

Kane610 commented:
Thanks a lot @MartinHjelmare !
